src/context.rs

```diff
@@ -352,15 +352,22 @@ impl<'buf> Secp256k1<AllPreallocated<'buf>> {
 
     /// Creates a context from a raw context.
     ///
+    /// The returned [`core::mem::ManuallyDrop`] context will never deallocate the memory pointed to
+    /// by `raw_ctx` nor destroy the context. This may lead to memory leaks. `ManuallyDrop::drop`
+    /// (or [`core::ptr::drop_in_place`]) will only destroy the context; the caller is required to
+    /// free the memory.
+    ///
     /// # Safety
-    /// This is highly unsafe, due to the number of conditions that aren't checked.
-    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer.
-    /// that was generated by *exactly* the same code/version of the libsecp256k1 used here.
-    /// * The capabilities (All/SignOnly/VerifyOnly) of the context *must* match the flags passed to libsecp256k1
-    /// when generating the context.
```

Comment:
This one still applies but is not mentioned in the new text.

Reply:
I can't remember why I removed this exactly because it was a long time ago but it seems such an obvious omission that I think me from the past would of had a reason. I'm out of my depth of knowledge here but I thought there was something about the libsecp context not differentiating any more? Are you sure we still want this line?

Reply:
From looking at `gen_new` I don't see that the libsecp context created is any different for the different All/SignOnly/VerifyOnly tags, or am I missing something?

Reply:
Ah, yeah, there is now no difference upstream between the capabilities, they're total no-ops. But we haven't updated the Rust API to reflect this yet because it'd be a big breaking change and we wanted to make sure upstream was finished changing first.

So I think this text is ok to drop.

Reply:
Sweet, my brain worked!

Reply:
> because it'd be a big breaking change

Maybe we could start with using `All` as the default type parameter and change the generic functions to accept any type parameter and deprecate the `All`, `SignOnly` and `VerifyOnly` types. This is not breaking but reflects it in the API.

> we wanted to make sure upstream was finished changing first.

Do we still need to? If there isn't certainty that this will stay, then we HAVE TO keep that requirement because an update would silently make some code unsound. And if we're certain this will stay we should at minimum state this in the docs and set `All` to be the default.

Reply:
>Do we still need to? 

No. Upstream had an actual release :).

>Maybe we could start with using All as the default type parameter and change the generic functions to accept any type parameter and deprecate the All, SignOnly and VerifyOnly types. This is not breaking but reflects it in the API.

Another strategy might be to stop exposing context objects in the API *at all*.

Reply:
We still need them for re-randomization, don't we?

Anyway, I prefer the API to be modified a little to not being modified at all because of the amount of work.

Reply:
In #388 (or somewhere) we came to a solution that enables re-randomization without needing explicit context objects.

Reply:
> Another strategy might be to stop exposing context objects in the API at all.

Would we use a global singleton for the context and leave the `secp256k1-sys` API the same?

Reply:
@tcharding yes, the secp-sys API would remain the same.